pkg/server/server.go

```diff
@@ -73,9 +77,33 @@ func (c *ProxyClientConnection) send(pkt *client.Packet) error {
 }
 
 func NewPendingDialManager() *PendingDialManager {
-	return &PendingDialManager{
+	pm := &PendingDialManager{
 		pendingDial: make(map[int64]*ProxyClientConnection),
 	}
+
+	// spawn a goroutine to verify connection waiting is timeout or not.
+	go func() {
+		ticker := time.NewTicker(3 * time.Second)
+		defer ticker.Stop()
```

Comment:
We should probably log V(2) something when this goroutine exits as it suggests that we will no longer be doing clean up.